src/librustc_resolve/lib.rs

```diff
@@ -191,10 +189,6 @@ fn resolve_struct_error<'b, 'a: 'b, 'c>(resolver: &'b Resolver<'a>,
                                         span: syntax_pos::Span,
                                         resolution_error: ResolutionError<'c>)
                                         -> DiagnosticBuilder<'a> {
-    if !resolver.emit_errors {
-        return resolver.session.diagnostic().struct_dummy();
```

Comment:
There *should* be a way to do speculative resolution without reporting errors.
It's used at least once now in fresh binding disambiguation and I'm using it in my error reporting branch as well.